### PR TITLE
Join hosts with packages

### PR DESCRIPTION
**Pull request by lpichler**
How to run:

```
curl "http://localhost:8080/content/packages" -H "Authorization: test_user;14" 
```

Authorization header:
is in format  <user_name>;<account_id>

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -83,26 +57,117 @@ func (c *PreFilterServer) GetContentPackages(ctx context.Context, request api.Ge
...
+func getIdentityFromContext(ctx context.Context) (user string, rhAccount int64, found bool) {
+	// Retrieve the value from the context
+	val := ctx.Value("user")
+	if val == nil {
+		// Handle case where the value is not found
+		return "", 0, false
+	}
+
+	// Convert the value to a string and split it
+	userInfo, ok := val.(string)
```
Comment (merlante) [inline server/server.go]: I thought that the type assertion `user, ok := ctx.Value("user").(string)` tested both the case where `ctx.Value("user")` was nil and also when it was not a string (so only one if statement needed).

File: server/server.go [hunk 2]
```diff
@@ -83,26 +57,117 @@ func (c *PreFilterServer) GetContentPackages(ctx context.Context, request api.Ge
...
+
+	// Convert the rhAccount part to int64
+	var err error
+	rhAccount, err = strconv.ParseInt(parts[1], 10, 64)
+	if err != nil {
+		// Handle error if the conversion fails
+		return "", 0, false
+	}
+
+	return user, rhAccount, true
```
Comment (merlante) [inline server/server.go:140]: Are you doing the parsing and splitting to extract user and org (which I assume is rhAccount)?

Comment (lpichler) [reply server/server.go:140]: yes, if I just did this way, on console we use xhr header like this:(it is base64 encoded)

```
curl -v "http://localhost:8000/api/rbac/v1/principals/" \-H "accept: application/json" \-H "x-rh-identity: eyJpZGVudGl0eSI6IHsiYWNjb3VudF9udW1iZXIiOiAiNjA4OTcxOSIsICJvcmdfaWQiOiIxMTc4OTc3MiIsICJ0eXBlIjogInVzZXIiLCAidXNlciI6IHsidXNlcm5hbWUiOiAiaW5zaWdodHMtcWEiLCAiZW1haWwiOiAiaW5zaWdodHMtcWFAcmVkaGF0LmNvbSIsICJmaXJzdF9uYW1lIjogIkd1bCIsICJsYXN0X25hbWUiOiAiRHVrYXQiLCAiaXNfYWN0aXZlIjogIlRydWUiLCAiaXNfb3JnX2FkbWluIjogIlRydWUiLCAiaXNfaW50ZXJuYWwiOiAiRmFsc2UiLCAibG9jYWxlIjogImVuX1VTIn0sICJpbnRlcm5hbCI6IHsib3JnX2lkIjogImRlZmF1bHRfcHJpbmNpcGFsIiwgImF1dGhfdHlwZSI6ICJpZGVudGl0eSJ9fSwgImVudGl0bGVtZW50cyI6IHt9fQ=="
```

but I think it is not needed - we just need user name and rh account

File: server/server.go (hunk 1 above)
Comment (lpichler) [reply server/server.go]: yes this is correct

Comment (lpichler) [reply server/server.go]: fixed

**Review by merlante — APPROVED** [review]

**Pull request merged**